src/selectors.js

```diff
@@ -17,6 +17,21 @@ Selectors = {
   getVmDisks ({ vmId }) {
     return getState().vms.getIn(['vms', vmId, 'disks'])
   },
+  getFirstCluster () {
+    return getState().clusters.get('clusters').toList().first()
+  },
+  getClusterById (id) {
+    return getState().clusters.get('clusters').toList().find(cluster =>
```

Comment:
If I read the reducers correctly, those entities (clusters/templates/oss) are already stored in ID-indexed hash-maps so I really not see the reason behind doing this `O(n)` lookup when a simple get by id from the immutable map should suffice.